Tried to shorten all sections (#269)
ITU-DevOps2026/ITU-minitwit: report/sections/01-4-current_state.md

Figure \ref{fig:current_state_codacy} shows the quality evolution of our codebase according to Codacy. We currently have the following types of issues: 

- **Security**, XX issues: Relates to a missing ValidateAntiForgeryToken.

- **Error prone**, XX issues: Relates to missing implementations of partial methods, unused variables and string operations.

> - **Compatibility**, XX issues: Use ANSI SQL syntax.

Reviewer: ```suggestion
- **Compatibility**, XX issues: Relates to not using ANSI SQL syntax.
```

![Codacy quality evolution of codebase \label{fig:current_state_codacy}](./images/current_state_codacy.png){width=80%}

### CodeQL

CodeQL scans our C# files and analyzes them for security vulnerabilities. Currently no security vulnerabilities have been found. 

### Trivy

> TrivyScanner scans our Dockerfiles for misconfigurations. Figure \ref{fig:current_state_trivy_misconfig} shows the report summary for misconfigurations in the latest scan. Currently, we have one misconfiguration that isn't fixed, located in our MySQL dockerfile which is run with root instead of a non-root user. From our initial research, this seems nontrivial to solve with the MySQL 8.0 base image that we have chosen, so currently the issue persists. 

Reviewer: ```suggestion
TrivyScanner scans our Dockerfiles for misconfigurations. Figure \ref{fig:current_state_trivy_misconfig} shows the report summary for misconfigurations in the latest scan. Currently, we have one unresolved misconfiguration, located in our MySQL dockerfile which is run with root as user. From our initial research, this seems nontrivial to solve with the MySQL 8.0 base image that we have chosen, so currently the issue persists. 
```

![Report from TrivyScanner on Dockerfile misconfigurations \label{fig:current_state_trivy_misconfig}](./images/current_state_trivy.png){width=60%}

> After scanning for misconfigurations TrivyScanner builds the Docker images and scans each image for vulnerabilities. Currently two of our images have vulnerabilities, minitwit-test and minitwit-mysql. Figure \ref{fig:current_state_trivy_test_vuln} shows the report of the vulnerabilities in the minitwit-test Docker image. Appendix [5.2](#appendix-for-trivyscanner-summary-of-minitwit-mysql-image-vulnerabilities) contains a summary of vulnerabilities found in the minitwit-mysql image. These issues stem from the MySQL 8.0 base image, and cannot be exploited, as long as our firewall holds, and no malicious user is able to access the database droplet. However, with more time, fixing these vulnerabilities would have been a priority, as they do pose a threat in the event that a malicious user gets access to the droplet.

Reviewer: ```suggestion
After scanning for misconfigurations TrivyScanner builds the Docker images and scans each image for vulnerabilities. Currently two of our images have vulnerabilities, minitwit-test and minitwit-mysql. Figure \ref{fig:current_state_trivy_test_vuln} shows the report of the vulnerabilities in the minitwit-test Docker image. Appendix [5.2](#appendix-for-trivyscanner-summary-of-minitwit-mysql-image-vulnerabilities) contains a summary of vulnerabilities found in the minitwit-mysql image. These issues stem from the MySQL 8.0 base image, currently unexploitable due to our firewall and restricted droplet access. With more time, fixing them would be a priority, as they do pose a threat if an attacker ever gains access.
```

![Report from TrivyScanner on Docker image minitwit-tests \label{fig:current_state_trivy_test_vuln}](./images/trivy_minitwit_test_vuln.png){width=100%}

> All reported issues reported by SonarQube Cloud, Codacy and Trivy should have been fixed and considered when they were set up. However, we've tried to fix issues continuously if they got reported during our CI pipeline for a pull request.

Reviewer: ```suggestion
All issues reported by SonarQube Cloud, Codacy and Trivy should have been fixed and considered when they were set up. However, we've tried to fix issues continuously if they got reported during our CI pipeline for a pull request. If we continued our work, the remaining issues would have to be adressed.
```

Reviewer: Is it on purpose this does not include the "if we had more time"-point anymore?
